server/src/main/scala/org/http4s/server/staticcontent/FileService.scala

```diff
@@ -84,11 +85,11 @@ object FileService {
   // Attempt to find a Range header and collect only the subrange of content requested
   private def getPartialContentFile[F[_]](file: File, config: Config[F], req: Request[F])(
       implicit F: Sync[F],
-      cs: ContextShift[F]): OptionT[F, Response[F]] =
-    OptionT.fromOption[F](req.headers.get(Range)).flatMap {
```

Comment:
I found the `OptionT.fromOption` followed by `flatMap` in here particularly misleading: it suggested there could be an effectful computation where there is not.